assets/RNAseq_report.Rmd

```diff
@@ -554,11 +556,18 @@ if (isProvided(params$path_relevel)) {
     }
 }
 
-# SizeFactors(cds) as indicator of library sequencing depth
-write.table(sizeFactors(cds),paste("differential_gene_expression/gene_counts_tables/sizeFactor_libraries.tsv",sep=""), append = FALSE, quote = FALSE, sep = "\t",eol = "\n", na = "NA", dec = ".", row.names = T,  col.names = F, qmethod = c("escape", "double"))
+# sizeFactors(cds) as indicator of library sequencing depth
+if (params$input_type == "featurecounts") {
+    sf <- sizeFactors(cds)
+} else {
+    # If tximport was used, sizeFactors doesn't work, so use workaround as described here: https://support.bioconductor.org/p/97676/
+    nm <- assays(cds)[["avgTxLength"]]
+    sf <- estimateSizeFactorsForMatrix(counts(cds)/nm)
+}
+write.table(sf,paste("differential_gene_expression/gene_counts_tables/sizeFactor_libraries.tsv",sep=""), append = FALSE, quote = FALSE, sep = "\t",eol = "\n", na = "NA", dec = ".", row.names = T,  col.names = F, qmethod = c("escape", "double"))
 
 # Write cds assay table to file
-write.table(counts(cds, normalized=T), paste("differential_gene_expression/gene_counts_tables/deseq2_table.tsv", sep=""), append=F, quote = F, sep = "\t", eol = "\n", na = "NA", dec = ".", row.names = T, col.names = T, qmethod = c("escape", "double"))
+write.table(counts(cds, normalized=T), paste0("differential_gene_expression/gene_counts_tables/deseq2_table.tsv"), append=F, quote = F, sep = "\t", eol = "\n", na = "NA", dec = ".", row.names = T, col.names = T, qmethod = c("escape", "double"))
```

Comment:
Could the output be named more explicitly? Like `deseq2_normalized_gene_counts.tsv` ?

Reply:
Renaming to `deseq2_library_scaled_gene_counts.tsv` to not produce confusion with the rlog/vst normalization stuff that is done later